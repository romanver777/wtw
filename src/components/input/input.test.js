/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import Input from "./input";

describe("Input component", () => {
  const props = {
    type: "password",
    placeholder: "Пароль",
    name: "user-password",
    id: "user-pass",
  };
  const Trenderer = TestRenderer.create(<Input {...props} />);

  it("renders correctly", () => {
    expect(Trenderer.toJSON()).toMatchSnapshot();
  });
  it("renders props type", () => {
    expect(Trenderer.root.findByType("input").props.type).toEqual(props.type);
  });
  it("renders props placeholder", () => {
    expect(Trenderer.root.findByType("input").props.placeholder).toEqual(
      props.placeholder
    );
  });
  it("renders props name", () => {
    expect(Trenderer.root.findByType("input").props.name).toEqual(props.name);
  });
  it("renders props id", () => {
    expect(Trenderer.root.findByType("input").props.id).toEqual(props.id);
  });
});
