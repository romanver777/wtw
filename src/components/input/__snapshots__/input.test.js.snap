// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<input
  id="user-pass"
  name="user-password"
  placeholder="Пароль"
  type="password"
/>
`;
